### PR TITLE
Allow another running TRex when run regression test through daemon se…

### PR DESCRIPTION
**Pull request by Cyborn-Minjae-Lee**
Hi Cisco team,

@egwakim @ejaecki @Kangphil @yonghwan007 @jsmoon @JunsubHan4rmEricsson @youngjun1 i @ejangle 

I would like to PR this feature.
This feature intend to enable running TRex in same server during regression test running without kill-running.
 
When we run regression test through daemon server, already running trex process are killed by regression test.
I change the code without “--kill-running” to run TRex in same server during regression test running.
After beginning regression test, the message will be printed “ TRex is already running. Use --kill-running flag to kill it.”

If you disagree with this,  I can add new option "--allow-another-trex" running TRex and Regression test at the same time.
How about you?

Best Regards,
Minjae Lee

### CODE REVIEW

**Comment by ybrustin** [comment]
The daemon server does not allow running several instances of TRex server.
Hiding error in client side will not help.

**Comment by Cyborn-Minjae-Lee** [comment]
I intend to use only 1 instance via daemon server for regression test, and other trex instance should no daemon at the same time.
How about you  adding new option "--allow-multi-instance" to restrict this usage ?

BR,
Minjae

**Comment by Cyborn-Minjae-Lee** [comment]
I add new option "--allow-multi-instance" to restrict this usage.

BR,
Minjae

File: scripts/automation/regression/trex_unit_test.py [hunk 1]
```diff
@@ -513,8 +516,8 @@ def begin (self):
         if not self.no_daemon:
             if self.kill_running:
                 client.kill_all_trexes()
-            elif client.get_trex_cmds():
-                print('TRex is already running. Use --kill-running flag to kill it.')
+            elif client.get_trex_cmds() and not self.allow_multi_instance:
```
Comment (ybrustin) [inline scripts/automation/regression/trex_unit_test.py]: Please rewrite as:
elif (not self.allow_multi_instance) and client.get_trex_cmds()
(for optimization)

**Comment by ybrustin** [comment]
Extra flag seems better option.

**Comment by Cyborn-Minjae-Lee** [comment]
Hi Yaroslav,

I have optimized coes using extra option "--allow-multi-instance".

**Comment by Cyborn-Minjae-Lee** [comment]
Hi Cisco team,

Please feedback for this PR ?

**Comment by ybrustin** [comment]
rebased to master, thanks.

**Pull request closed without merging**